src/pac-file-tester.ts

```diff
@@ -30,11 +30,8 @@ export const testPacFile = async (file: string, url: string, ip: string = addres
 }
 
 export const getHost = (url: string): string => {
-  const regex = /^(?:https?:\/\/)?(?:[^@\n]+@)?(?:www\.)?([^:\/\n]+)/im
-
-  const match = regex.exec(url)
-
-  return match[1]
+  const urlObj = new URL(url);
+  return urlObj.hostname
```

Comment:
Not sure why the original strips `www.` but pretty sure we should be checking against the actual hostname, easy to obtain that with built-in URL class.